hera_mc/cm_hookup.py

```diff
@@ -697,28 +396,31 @@ def read_hookup_cache_from_file(self):
         """
         if os.path.exists(self.hookup_cache_file):
             with open(self.hookup_cache_file, 'r') as outfile:
-                save_dict = json.load(outfile)
-
-            self.cached_at_date = Time(save_dict['at_date_gps'], format='gps')
-            self.cached_hookup_type = save_dict['hookup_type']
-            self.cached_hookup_list = save_dict['hookup_list']
+                cache_dict = json.load(outfile)
+            if self.hookup_cache_file_OK(cache_dict):
+                print("<<<Cache IS current with database>>>")
+            else:
+                print("<<<Cache is NOT current with database>>>")
+            self.cached_at_date = Time(cache_dict['at_date_gps'], format='gps')
+            self.cached_hookup_type = cache_dict['hookup_type']
+            self.cached_hookup_list = cache_dict['hookup_list']
             hookup_dict = {}
-            for key, value in six.iteritems(save_dict['hookup_dict']):
-                # this should only contain dicts made from HookupDossierEntry
+            for key, value in six.iteritems(cache_dict['hookup_dict']):
+                # this should only contain dicts made from HookupEntry
                 # add asserts to make sure
                 assert(isinstance(value, dict))
                 assert(sorted(value.keys()) == sorted(['entry_key', 'hookup', 'fully_connected',
                                                        'hookup_type', 'columns', 'timing', 'sysdef']))
-                hookup_dict[key] = HookupDossierEntry(input_dict=value)
+                hookup_dict[key] = cm_dossier.HookupEntry(input_dict=value)
 
             self.cached_hookup_dict = hookup_dict
-            self.part_type_cache = save_dict['part_type_cache']
+            self.part_type_cache = cache_dict['part_type_cache']
         else:
-            self._hookup_cache_to_use(force_new_cache=True)
-            self.read_hookup_cache_from_file()
+            print("No cache file.  Returning None.")
```

Comment:
Should this be a warning or error rather than a print?

Reply:
Took out file exists check, so it will raise the appropriate error.